lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1916,284 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
+    vault_name: str,
+) -> Any:
+    keyvault_client = get_key_vault_management_client(platform)
+
+    # Fetch the current policies and add the VM's policy
+    vault = keyvault_client.vaults.get(resource_group_name, vault_name)
+    current_policies = vault.properties.access_policies
+    new_policy = AccessPolicyEntry(
+        tenant_id=tenant_id,
+        object_id=object_id_vm,
+        permissions=Permissions(keys=["all"], secrets=["all"], certificates=["all"]),
+    )
+    current_policies.append(new_policy)
+
+    # Update the vault with the new policies
+    vault.properties.access_policies = current_policies
+    keyvault_poller = keyvault_client.vaults.begin_create_or_update(
+        resource_group_name, vault_name, vault
+    )
+
+    return keyvault_poller.result()
+
+
+def delete_keyvault(
+    platform: "AzurePlatform", vault_name: str, log: Logger, resource_group_name: str
+) -> None:
+    keyvault_client = get_key_vault_management_client(platform)
+
+    keyvault_poller = keyvault_client.vaults.delete(resource_group_name, vault_name)
+    log.debug(f"{keyvault_poller} - Key Vault {vault_name} deleted successfully.")
+
+
+def get_certificate_client(
+    vault_url: str, platform: "AzurePlatform"
+) -> CertificateClient:
+    return CertificateClient(vault_url, platform.credential)
+
+
+def get_secret_client(vault_url: str, platform: "AzurePlatform") -> SecretClient:
+    return SecretClient(vault_url, platform.credential)
+
+
+def create_certificate(
+    platform: "AzurePlatform",
+    vault_url: str,
+    log: Logger,
+    cert_name: str,
+    retries: int = 5,
+    delay: int = 2,
+) -> str:
+    certificate_client = get_certificate_client(vault_url, platform)
+    secret_client = get_secret_client(vault_url, platform)
+
+    cert_policy = CertificatePolicy.get_default()
+
+    for _attempt in range(retries):
+        # Create certificate
+        create_certificate_result = certificate_client.begin_create_certificate(
+            cert_name, policy=cert_policy
+        )
+        log.info(
+            f"Certificate '{cert_name}' has been created. "
+            f"Result: {create_certificate_result}"
+        )
+        certificate_client.update_certificate_properties(
+            certificate_name=cert_name, enabled=True
+        )
+        break
+
+    secret_id: Optional[str] = secret_client.get_secret(name=cert_name).id
+    if secret_id:
+        secret_url_without_version = secret_id.rsplit("/", 1)[0]
+        return secret_url_without_version
+    else:
+        raise LisaException(f"Failed to retrieve secret ID:'{cert_name}'.")
+
+
+def check_certificate_existence(
+    vault_url: str, cert_name: str, log: Logger, platform: "AzurePlatform"
+) -> bool:
+    certificate_client = CertificateClient(
+        vault_url=vault_url, credential=platform.credential
+    )
+
+    try:
+        certificate = certificate_client.get_certificate(cert_name)
+        log.info(f"Cert found '{certificate.name}'")
```

Comment:
Move it to debug level log. It doesn't need to be info level.

Reply:
There are other similar places. info level log is to tell a story, debug level is to provide details. In most cases, only test case level needs a few info level logs.